### PR TITLE
Enable markdownlint `no-inline-html`

### PR DESCRIPTION
**Pull request by silverwind**
With a few exceptions to accomodate current usage. [Rule docs](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md033---inline-html).

### CODE REVIEW

**Review by yardenshoham — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -344,7 +344,7 @@ lint-frontend: node_modules
 	npx eslint --color --max-warnings=0 --ext js,vue web_src/js build *.config.js docs/assets/js tests/e2e
 	npx stylelint --color --max-warnings=0 web_src/css
 	npx spectral lint -q -F hint $(SWAGGER_SPEC)
-	npx markdownlint docs *.md
+	npx markdownlint --ignore CHANGELOG.md docs *.md
```
Comment (silverwind) [inline Makefile]: Changelog has a few violations from the commit messages, I felt it's not worth trying to fix those so I ignored the file.

File: .markdownlintignore [hunk 2]
```diff
@@ -0,0 +1 @@
+/CHANGELOG.md
```
Comment (silverwind) [inline .markdownlintignore:1]: Changelog had a few violations stemming from commit message, so these should not be fixed.

**Comment by silverwind** [comment]
Note that the rule is a bit too harsh in some cases, that's why the exception list is so long. For example to middle-align something, one can do:

````
<p align="center">
  I'm <b>bold</b>
</p>
````
In such a case, `<b>` usage is unavoidable and okay, but it would trigger the linter. So I'm unsure whether we should really enable it.

**Comment by silverwind** [comment]
I think I'll decide against this. It could be enabled of there is a smarter rule that detects HTML attribute usage, but as it is, the rule is just too crude to cover our use cases.

**Pull request closed without merging**